### ISSUE TITLE
Giving an Item Drone Ancient Scepter bugs the run with any amount of allied units

### ISSUE THREAD

**Issue by SalTheThief** [ThinkInvis/RoR2-TinkersSatchel#67]
### Describe the Bug
TL;DR Since Ancient Scepter rerolls itself to another red item when given to a unit without an upgrade it leads to an infinite amount of Happiest Mask ghosts.
Ancient Scepter mod here: https://thunderstore.io/package/amogus_lovers/StandaloneAncientScepter/1.1.1/

1. Get an Ancient Scepter
2. Give the Ancient Scepter to an Item Drone
3. Your allied units, such as drones and beetle guards, do not have an Ancient Scepter upgrade, so it will give them a different red item instead.
4. Eventually one will get a Happiest Mask, which will spawn allied ghosts, which do not have an Ancient Scepter upgrade, so they get a red item, which will eventually be another Happiest Mask... you can see where this becomes a problem.

### Expected Behavior
Blacklist Ancient Scepter from giving it to allied units (but not allied survivors).

**Comment by ThinkInvis** [comment]
Possibly resolved by f8ece16; pending testing.

**Comment by ThinkInvis** [comment]
Resolved -- Item Drones should no longer be able to accept Ancient Scepter in the repair prompt, as originally intended. Also confirmed that the bug did occur before this fix.
